commons/src/test/java/io/aiven/kafka/connect/common/source/task/PartitionInFilenameDistributionStrategyTest.java

```diff
@@ -0,0 +1,138 @@
+
+import org.apache.kafka.common.config.ConfigException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+final class PartitionInFilenameDistributionStrategyTest {
+
+    @Test
+    void partitionInFileNameDefaultAivenS3Sink() {
+        final SourceObjectDistributionStrategy taskDistribution = new PartitionInFilenameDistributionStrategy(2,
+                "logs-\\{\\{partition}}-\\{\\{start_offset}}");
+        assertThat(taskDistribution.isPartOfTask(1, "logs-1-00112.gz")).isTrue();
+    }
+
+    @Test
+    void partitionLocationNotSetExpectException() {
+        assertThatThrownBy(() -> new PartitionInFilenameDistributionStrategy(2, "logs-23-<partition>-<start_offset>"))
+                .isInstanceOf(ConfigException.class)
+                .hasMessage(
+                        "Partition pattern {{partition}} not found, please configure the expected source to include the partition pattern.");
+
+    }
+
+    @ParameterizedTest
+    @CsvSource({ "logs-\\{\\{partition}}-\\{\\{start_offset}},logs-0-00112.gz",
+            "logs-2024-\\{\\{timestamp}}-\\{\\{partition}}-\\{\\{start_offset}},logs-2024-20220201-0-00112.gz",
+            "logs-2023-\\{\\{partition}}-\\{\\{start_offset}},logs-2023-0-00112.gz",
+            "logs1-\\{\\{timestamp}}-\\{\\{timestamp}}-\\{\\{timestamp}}-\\{\\{partition}}-\\{\\{start_offset}},logs1-2022-10-02-10-00112.gz",
+            "8952\\{\\{partition}}-\\{\\{start_offset}},89521-00112.gz",
+            "Emergency-TEST\\{\\{partition}}-\\{\\{start_offset}},Emergency-TEST1-00112.gz",
+            "PROD-logs-\\{\\{partition}}-\\{\\{start_offset}},PROD-logs-1-00112.gz",
+            "DEV_team_\\{\\{partition}}-\\{\\{start_offset}},DEV_team_1-00112.gz",
+            "DEV_team_\\{\\{partition}}-\\{\\{start_offset}},DEV_team_1-00112.gz",
+            "timeseries-\\{\\{partition}}-\\{\\{start_offset}},timeseries-1-00112.gz" })
+    void testPartitionFileNamesAndExpectedOutcomes(final String configuredFilename, final String filename) {
+        final SourceObjectDistributionStrategy taskDistribution = new PartitionInFilenameDistributionStrategy(1,
+                configuredFilename);
+        // This test is testing the filename matching not the task allocation.
+        assertThat(taskDistribution.isPartOfTask(0, filename)).isTrue();
+    }
+
+    @ParameterizedTest
+    @CsvSource({ "different-topic-\\{\\{partition}}-\\{\\{start_offset}},logs-1-00112.gz",
+            "no-seperator-in-date-partition-offset-\\{\\{timestamp}}-\\{\\{partition}}-\\{\\{start_offset}},no-seperator-in-date-partition-offset-202420220201100112.gz",
+            "logs-2024-\\{\\{timestamp}}-\\{\\{partition}}-\\{\\{start_offset}},logs-20201-1-00112.gz",
+            "logs-2024-\\{\\{timestamp}}\\{\\{partition}}-\\{\\{start_offset}},logs-202011-00112.gz",
+            "logs-2023-\\{\\{partition}}-\\{\\{start_offset}},logs-2023-one-00112.gz" })
```

Comment:
can we add any edge cases like 

        "empty-pattern,,",
        "logs-\\{\\{partition}}-\\{\\{start_offset}},",
        ",logs-1-00112.gz"